### PR TITLE
EVEREST-107 Remove kube-rbac-proxy and update kubebuilder project to go/v4

### PR DESCRIPTION
**Pull request by recharte**
**Remove kube-rbac-proxy and update kubebuilder project to go/v4**
---
**Problem:**
EVEREST-107

[GCR will die soon](https://cloud.google.com/artifact-registry/docs/transition/prepare-gcr-shutdown). Therefore, we need to remove kube-rbac-proxy ASAP.

**Solution:**
As per [kubebuilder's guidance](https://book.kubebuilder.io/reference/metrics.html?highlight=kube-rbac-proxy#important-if-you-are-using-kube-rbac-proxy) we should move the project to a go/v4 base which no longer uses kube-rbac-proxy.

**CHECKLIST**
---
**Helm chart**
- [x] Is the [helm chart](https://github.com/percona/percona-helm-charts/tree/main/charts/everest) updated with the new changes? (if applicable)

**Jira**
- [ ] Is the Jira ticket created and referenced properly?

**Tests**
- [ ] Is an Integration test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?

### CODE REVIEW

**Comment by recharte** [comment]
@mayankshah1607 I think I'll need your assistance in translating these changes to the helm chart :pray:

**Review by michal-kralik — APPROVED** [review]

File: deploy/bundle.yaml [hunk 1]
```diff
@@ -1488,67 +1739,15 @@ spec:
...
         - /manager
-        env:
-        - name: SYSTEM_NAMESPACE
-          valueFrom:
-            fieldRef:
-              fieldPath: metadata.namespace
-        - name: POD_NAME
-          valueFrom:
-            fieldRef:
-              fieldPath: metadata.name
```
Comment (recharte) [inline deploy/bundle.yaml:1551]: Looks like I messed up a bit :sweat_smile:  need to get these back...

Comment (mayankshah1607) [inline deploy/bundle.yaml:1551]: Why are we getting rid of this? Its used in the DBEngine controller

File: deploy/bundle.yaml [hunk 2]
```diff
@@ -1488,67 +1739,15 @@ spec:
...
+        - --metrics-bind-address=:8443
         - --leader-elect
+        - --health-probe-bind-address=:8081
         command:
         - /manager
-        env:
-        - name: SYSTEM_NAMESPACE
-          valueFrom:
-            fieldRef:
-              fieldPath: metadata.namespace
```
Comment (mayankshah1607) [inline deploy/bundle.yaml:1547]: If we're removing this, we need to specify `--system-namespace` flag.. Its done in the Helm chart, but this manifest will result in a failing operator deployment

Comment (recharte) [reply deploy/bundle.yaml:1547]: Just found this issue 1min ago: https://github.com/percona/everest-operator/pull/597#discussion_r1863307391

File: deploy/bundle.yaml (hunk 1 above)
Comment (recharte) [reply deploy/bundle.yaml:1551]: Just found this issue 1min ago: https://github.com/percona/everest-operator/pull/597#discussion_r1863307391

**Comment by recharte** [comment]
https://github.com/percona/percona-helm-charts/pull/436

**Comment by camilamacedo86** [comment]
Hi @recharte 

Terrific work !!! 🚀 
Thank you for upgrading your project 🥇

**Pull request merged**